docs/source/basics/customizing_interface.md

```diff
@@ -148,3 +109,292 @@ Each key in `metadata.parameters` must be the name of one of the parameters of t
+
+1. LatchFile
+
+```python
+from latch.types import LatchFile
+from typing import Optional
+
+...
+@workflow
+def bactopia_wf(
+    ...
+    fastq_one: Optional[LatchFile] = None,
+    ...
+)
+```
+![LatchFile](../assets/ui/optional-latch-file.png)
+
+`LatchFile` receives a button that allows users of the workflow to select data from their Latch account. The `Optional` type renders the toggle for `fastq_one`. When the toggle is turned on, Latch automatically detects the empty path and throws a warning. Additionally, you can set the default value to the path to `None`.
+
+2. LatchDir
+```python
+from latch.types import LatchDir
+
+...
+@workflow
+def bactopia_wf(
+    ...
+    output_dir: LatchDir,
+    ...
+)
+```
+![LatchDir](../assets/ui/latchdir.png)
+
+3. Boolean
+```python
+hybrid: bool = False,
+```
+![Boolean](../assets/ui/boolean.png)
+
+4. Enum
+```python
+from enum import Enum
+
+# You must define your Enum as a python class before using it as an annotation.
+class SpeciesGenomeSize(Enum):
+    mash = "mash estimate"
+    min = "min"
+    median = "median"
+    mean = "mean"
+    max = "max"
+
+...
+@workflow
+def bactopia_wf(
+    ...
+    species_genome_size: SpeciesGenomeSize,
+    ...
+)
+```
+
+![Enum](../assets/ui/enum.png)
+
+5. Int 
+```python
+@workflow
+def bactopia_wf(
+    ...
+    coverage: int,
+    ...
+)
+```
+
+![int](../assets/ui/int.png)
+
+6. Str
+```python
+@workflow
+def bactopia_wf(
+    ...
+    sample_name: str = "sample1"
+    ...
+)
+```
+
+![str](../assets/ui/str.png)
+
+7. List 
+```python
+from latch.types import LatchFile
+from typing import List
+
+@workflow
+def rnaseq(
+    sample_identifiers: List[str], 
+    sample_reads: List[LatchFile]
+): 
+...
+```
+
+![List](../assets/ui/list.png)
+
+When `List` is used, Latch generates a plus sign, where users can add additional values of the same type. For `LatchFile`s specifically, an additional button **Bulk Add Files** is generated, allowing users to select multiple files at once. 
+
+
+
+8. Dataclass
+
+If you want to handle file references and their associated metadata as an input to your workflow, you may want to use a `dataclass`.
+
+```python
+from dataclasses import dataclass
+from dataclasses_json import dataclass_json
+
+@dataclass_json
+@dataclass
+class Sample: 
+    name: str
+    fastq: LatchFile
+
+@workflow
+def rnaseq(
+    samples: List[Sample]
+):
+
+```
+
+![List of class](../assets/ui/dataclass.png)
+
+Here, we are passing a list of `Sample`s as the input. On the Latch interface, when a user clicks the `+ Sample` button, a new block will be added with two parameters of the Python class `name` and `fastq`.
+
+---
+## Adding Documentation to your Workflow
+While most of the metadata of a workflow will be encapsulated in a LatchMetadata object, we still require a docstring in the body of the workflow function which specifies both a short and long-form description.
+
+
+
+### One Line Description
```

Comment:
Shouldn't this description stuff all just be contained at the top in one place vs being repeated?